libs/Wallet/src/NewWalletAccountController.cpp

```diff
@@ -0,0 +1,188 @@
+#include "Status/Wallet/NewWalletAccountController.h"
+
+#include <StatusGo/Wallet/WalletApi.h>
+
+#include <StatusGo/Accounts/AccountsAPI.h>
+#include <StatusGo/Accounts/Accounts.h>
+#include <StatusGo/Accounts/accounts_types.h>
+#include <StatusGo/Metadata/api_response.h>
+#include <StatusGo/Utils.h>
+#include <StatusGo/Types.h>
+
+#include <Onboarding/Common/Constants.h>
+
+#include <QQmlEngine>
+#include <QJSEngine>
+
+#include <ranges>
+
+namespace GoAccounts = Status::StatusGo::Accounts;
+namespace GoWallet = Status::StatusGo::Wallet;
+namespace UtilsSG = Status::StatusGo::Utils;
+namespace StatusGo = Status::StatusGo;
+
+namespace Status::Wallet {
+
+NewWalletAccountController::NewWalletAccountController(std::shared_ptr<Helpers::QObjectVectorModel<WalletAccount>> accounts)
+    : m_accounts(accounts)
+    , m_mainAccounts(std::move(filterMainAccounts(*accounts)), "account")
+    , m_derivedAddress("derivedAddress")
+    , m_derivationPath(Status::Constants::General::PathWalletRoot)
+{
+}
+
+NewWalletAccountController::~NewWalletAccountController()
+{
+}
+
+QAbstractListModel* NewWalletAccountController::mainAccountsModel()
+{
+    return &m_mainAccounts;
+}
+
+QAbstractItemModel *NewWalletAccountController::currentDerivedAddressModel()
+{
+    return &m_derivedAddress;
+}
+
+QString NewWalletAccountController::derivationPath() const
+{
+    return m_derivationPath.get();
+}
+
+void NewWalletAccountController::setDerivationPath(const QString &newDerivationPath)
+{
+    if (m_derivationPath.get() == newDerivationPath)
+        return;
+    m_derivationPath = GoAccounts::DerivationPath(newDerivationPath);
+    emit derivationPathChanged();
+
+    auto oldCustom = m_customDerivationPath;
+    auto found = searchDerivationPath(m_derivationPath);
+    m_customDerivationPath = std::get<0>(found) == nullptr;
```

Comment:
`const auto &[foo, bar] = searchDerivationPath(m_derivationPath);` instead of repeating `std::get<>()` below